internal/dataapi/listener_test.go

```diff
@@ -208,6 +208,27 @@ func TestSmokeDataAPI(t *testing.T) {
 		require.NoError(t, err)
 		assert.JSONEq(t, `{"documents":[`+docs[2]+`]}`, string(body))
 	})
+
+	t.Run("InsertOneEmptyId", func(t *testing.T) {
+		jsonBody := `{
+			"database": "` + db + `",
+			"collection": "` + coll + `",
+			"document": ` + `{"v":"foo"}` + `
+		}`
+
+		res, err := postJSON(t, "http://"+addr+"/action/insertOne", jsonBody)
+		require.NoError(t, err)
+		assert.Equal(t, http.StatusOK, res.StatusCode)
+
+		body, err := io.ReadAll(res.Body)
+		require.NoError(t, err)
+
+		bodyStr := strings.TrimSpace(string(body))
+		bodyStr = strings.TrimPrefix(bodyStr, `{"insertedId":"`)
+		bodyStr = strings.TrimSuffix(bodyStr, `"}`)
+
+		assert.Regexp(t, `^[0-9a-f]{24}$`, bodyStr, "expected %q, got %q", `{"insertedId":"<ObjectID>"} (%s)`, string(body))
```

Comment:
If you are asserting with regex, why not match entire body `{"insertedId":"[0-9a-f]{24}"}`?